### PR TITLE
Issue #378: html self contained changes

### PR DESCRIPTION
**Pull request by shachiagarwal**
# Please include a summary of the changes

vane/tests_client.py - create self contained html file if self contained option is specified in the definitions.yaml
Other changes - the date timestamp of html file now exactly matches docx file.  

# Any specific logic/part of code you need extra attention on
None 

Explain any specific logic you need special attention on

# Include the Issue number and link
#378 
Make sure to link the issue in the github PR UI 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [X] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [X] This change requires a documentation update
- - [ ] Other (please specify)

# Effort required on reviewers end

- - [X] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [X] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update
Getting started guide needs to updated. Opened this issue https://github.com/aristanetworks/vane/issues/518
    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: vane/utils.py [hunk 1]
```diff
@@ -114,3 +114,15 @@ def now():
     """Return current date and time"""
 
     return (datetime.datetime.now()).strftime("%d-%m-%Y %H:%M:%S")
+
+
+def return_date():
+    """Genreate a formatted date and return to calling
```
Comment (rewati-arista) [inline vane/utils.py:120]: typo
```suggestion
    """Generate a formatted date and return to calling
```

File: tests/unittests/fixtures/expected_test_host.json [hunk 2]
```diff
@@ -1 +1 @@
-{"tests/unittests/fixtures/host/test_host.py": ["01/01/2023 00:00:00", "TD: Verify hostname is set on device is correct", "TS: Collecting the output of 'show hostname' command from DUT", "TS: Verify LLDP system name", "TS: Creating test report based on results"]}
\ No newline at end of file
+{"tests/unittests/fixtures/host/test_host.py": ["September 07, 2023 06:22:55PM", "TD: Verify hostname is set on device is correct", "TS: Collecting the output of 'show hostname' command from DUT", "TS: Verify LLDP system name", "TS: Creating test report based on results"]}
```
Comment (rewati-arista) [inline tests/unittests/fixtures/expected_test_host.json:1]:
```suggestion
{"tests/unittests/fixtures/host/test_host.py": ["September 07, 2023 06:22:55PM", "TD: Verify hostname set on device is correct", "TS: Collecting the output of 'show hostname' command from DUT", "TS: Verify LLDP system name", "TS: Creating test report based on results"]}
```

File: tests/unittests/fixtures/expected_test_host.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 test_host
 =========
   
-Date generated: 01/01/2023 00:00:00
+Date generated: September 07, 2023 06:22:55PM
 #  Verify hostname is set on device is correct
```
Comment (rewati-arista) [inline tests/unittests/fixtures/expected_test_host.md:6]:
```suggestion
#  Verify hostname set on device is correct
```

File: tests/unittests/test_tests_step_client.py [hunk 4]
```diff
@@ -10,14 +10,14 @@
 TEST_DIR = "tests/unittests/fixtures/host"
 TEST_FILE = ["tests/unittests/fixtures/host/test_host.py"]
 TEST_STEP = [
-    "01/01/2023 00:00:00",
+    "September 07, 2023 06:22:55PM",
     "TD: Verify hostname is set on device is correct",
```
Comment (rewati-arista) [inline tests/unittests/test_tests_step_client.py:14]:
```suggestion
    "TD: Verify hostname set on device is correct",
```

**Review by rewati-arista — APPROVED** [review]
LGTM. A couple of typos which had been introduced earlier, flagged them here, upto your discretion to handle them or not, logical changes look good hence approving🙌🏻

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**